APIs/controllers/authCtrl.js

```diff
@@ -225,8 +225,175 @@ async function register(req, res) {
+                                    if (err) {
+                                        console.error(err);
+                                        return res.status(500).json({
+                                            success: false,
+                                            message: err.message
+                                        });
+                                    }
+                                    if(data){
+                                        return res.status(200).json({
+                                            success: true,
+                                            message: "Password Updated Successfully"
+                                        });
+                                    }
+                                })
+                            }
+                        })
+                    }
+                    else {
+                        console.error(err);
+                        return res.status(500).json({
+                            success: false,
+                            message: "Invalid user"
+                        });
+                    }
+                })
+            }else{
+                return res.status(500).json({
+                    success: false,
+                    message: "Token expired! Try again!"
+                });
+            }
+        }
+        else{
+            return res.status(500).json({
+                success: false,
+                message: "Invalid Token"
+            });
+        }
+    } )
+}
 
+function forget(req, res){
+    try{
+        if (!req.body.email_id) {
+            res.status(500).json({
+                success: false,
+                message: "email_id is required"
+            });
+            return;
+        }
+        db.Users.findOne({
+            email_id: req.body.email_id
+        }, (err, user) => {
+
+            if (err) {
+                console.error(err);
+                return res.status(500).json({
+                    success: false,
+                    message: err.message
+                });
+            }
 
+            if (user) {
+                pass_reset = {
+                    reset_token: crypto.randomBytes(16).toString('hex'),
+                    user_id: user._id
+                }
+                pass_obj = new db.PasswordReset(pass_reset);
+                pass_obj.save( (err, data) => {
+                    if (err) {
+                        console.error(err);
+                        return res.status(500).json({
+                            success: false,
+                            message: err.message
+                        });
+                    }
+                    if (data){
+                        const html = `Hi there,
+        <br/>
+        Go to the Link below to Reset Your Password! Link valid for 30 minutes.
+        <br/><br/>
+        Please verify your email <a href="http://localhost:4000/api/v1/reset?id=${data.reset_token}">HERE</a>
```

Comment:
@AyonPal hey buddy!
The link your sending in the mail won't be valid as your reset controller is looking for both `reset_token & new_pass` in the body, as I can see you're sending a direct API request with `reset_token` in parameters, instead, send the reset_token as a text and link to a frontend page (not yet made) where user can paste his `reset_token & new_pass` and get his password reset.

Reply:
Should i set the token like 6 digit numeric or something  in the email?